New pages: CSS background-repeat x and y (#42184)
mdn/content: files/en-us/web/css/reference/properties/background-repeat-x/index.md

{{InteractiveExample("CSS Demo: background-repeat-x")}}

```css interactive-example-choice
background-repeat-x: repeat;
```

```css interactive-example-choice
background-repeat-x: space;
```

```css interactive-example-choice
background-repeat-x: round;
```

```css interactive-example-choice
background-repeat-x: no-repeat;
```

```html interactive-example
<section id="default-example">
  <div id="example-element"></div>
</section>
```

```css interactive-example
#example-element {
  background: #cccccc url("/shared-assets/images/examples/moon.jpg") no-repeat
    center / 120px;
  min-width: 100%;
  min-height: 100%;
}
```

## Syntax

```css
/* Keyword values */
background-repeat-x: repeat;
background-repeat-x: space;
background-repeat-x: round;
background-repeat-x: no-repeat;

/* Global values */
background-repeat-x: inherit;
background-repeat-x: initial;
background-repeat-x: revert;
background-repeat-x: revert-layer;
background-repeat-x: unset;
```

The `background-repeat-x` property is specified as one or more values, separated by commas.

## Values

- `repeat`

>   - : The default value. The image is repeated as many times as needed to cover the entire width of the background image painting area, with the edge image being clipped if the width of the painting area is not a multiple of the width of your background image.

Reviewer: ```suggestion
  - : The default value. The image is repeated as many times as needed to cover the entire width of the background image painting area. The edge image is clipped if multiple images do not fit exactly into the background width.
```

- `no-repeat`
  - : The image is not repeated. The position of the non-repeated background image is defined by the {{cssxref("background-position")}} CSS property.

- `space`
  - : The image is repeated as much as possible without clipping. The first and last images are pinned to the left and right of the element, and whitespace is distributed evenly between the images. The {{cssxref("background-position-x")}} property is ignored unless only one image can be displayed without clipping. The only case where clipping happens using `space` is when the width of the image is wider than the element, so there isn't enough room to display one image.

- `round`

>   - : As the allowed space increases in size, the repeated images will stretch (leaving no gaps) until there is room for another one to be added. If If the image doesn't fit a whole number of times, it is rescaled so that it does.

Reviewer: ```suggestion
  - : The image is repeated horizontally. As the available space increases in size, the repeated images will stretch (leaving no gaps) until there is room for another one to be added. If multiple images don't fit the background exactly, they are rescaled to fit.
```

## Description

> The `background-repeat-x` property accepts a comma-separated list of [`<repetition`](#values) keyterms, which define how the background image(s) should repeat horizontally, or not repeat at all.

Reviewer: ```suggestion
The `background-repeat-x` property accepts a comma-separated list of [`<repetition>`](#values) keyterms, which define how the background image(s) should repeat horizontally, or not repeat at all.
```

Alternatively, the repeated background image can be stretched to cover the entire width of the background area without clipping. With `round`, the repeated image is stretched to fill all the available space until there is room to add an additional repeated image. For example, given a background image that is 100px x 100px and a background paint area that is 1099px wide, the image will be repeated 10 times in the horizontal direction with each image stretched in to be 109.9px wide, possibly altering the images {{glossary("aspect ratio")}} and distorting it. If the width of the paint area increases by 1px, becoming 1100px wide, an 11th image will fit horizontally, with each image being painted at 100px wide, and no longer stretched in the horizontal direction.

## Formal definition

{{cssinfo}}

## Formal syntax

{{csssyntax}}

## Examples

> ### Setting background-repeat-x

Reviewer: I've not tested this as it doesn't work on any browser I have available. I'm assuming it works.
